### PR TITLE
Fix headers

### PR DESCRIPTION
**Pull request by yoavst**
Beside making few `var`s to `val`s, it fixes the `content-type` problem.

### CODE REVIEW

File: fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Encoding.kt [hunk 1]
```diff
@@ -74,9 +75,13 @@ class Encoding : Fuel.RequestConvertible {
     private fun queryFromParameters(params: List<Pair<String, Any?>>?): String {
         return params?.let {
             params.filterNot { it.second == null }
-                    .mapTo(arrayListOf<String>()) { "${it.first}=${it.second}" }
+                    .mapTo(ArrayList()) { "${it.first}=${it.second}" }
```
Comment (kittinunf) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Encoding.kt]: We should stick with Kotlin types. Using `mutableListOf()` should be better option.

File: fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Request.kt [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Request : Fuel.RequestConvertible {
     lateinit var url: URL
     var httpBody: ByteArray = ByteArray(0)
 
-    var httpHeaders = hashMapOf<String, String>()
+    val httpHeaders: MutableMap<String, String> = hashMapOf<String, String>()
```
Comment (kittinunf) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Request.kt]: Explicit type declaration is not needed.

File: fuel/src/main/kotlin/com/github/kittinunf/fuel/core/requests/AsyncTaskRequest.kt [hunk 3]
```diff
@@ -9,21 +9,22 @@ class AsyncTaskRequest(val task: TaskRequest) : TaskRequest(task.request) {
     var failureCallback: ((FuelError, Response) -> Unit)? = null
 
     override fun call(): Response {
-        try {
+        return try {
```
Comment (kittinunf) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/core/requests/AsyncTaskRequest.kt]: Nice 👍

**Comment by kittinunf** [comment]
If you using rebase against `master`, this commit (737e7c5) is not needed.

File: fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Request.kt (hunk 2 above)
Comment (yoavst) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Request.kt]: Yes, it could be replaced with `mutableMapOf`, I want it to be kotlin type.

Comment (kittinunf) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Request.kt]: Yeah, that would be ideal. Can you help me fix this as well?

**Comment by kittinunf** [comment]
It all looks good, after you fix my concerns.

**Comment by kittinunf** [comment]
OK .. you are GTG.

**Pull request merged**